include/ginkgo/core/distributed/matrix.hpp

```diff
@@ -388,6 +408,37 @@ class Matrix
      * @return  this.
      */
     Matrix& operator=(Matrix&& other);
+
+    std::vector<comm_index_type> get_recv_sizes() const override
+    {
+        return recv_sizes_;
+    };
+    std::vector<comm_index_type> get_send_sizes() const override
+    {
+        return send_sizes_;
+    };
+    std::vector<comm_index_type> get_recv_offsets() const override
+    {
+        return recv_offsets_;
+    };
+    std::vector<comm_index_type> get_send_offsets() const override
+    {
+        return send_offsets_;
+    }
+    array<local_index_type> get_gather_idxs() const override
```

Comment:
Just for my understanding, why don't we return  `const array<local_index_type>&` ?

Reply:
After discussing with Marcel, we decide to use `friend`.